### PR TITLE
[Merged by Bors] - feat: port SetTheory.Game.Nim

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Game/Nim.lean [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+import Mathlib.SetTheory.Game.Birthday
+import Mathlib.SetTheory.Game.Impartial
+
+/-!
+# Nim and the Sprague-Grundy theorem
+
+This file contains the definition for nim for any ordinal `o`. In the game of `nim o₁` both players
+may move to `nim o₂` for any `o₂ < o₁`.
+We also define a Grundy value for an impartial game `G` and prove the Sprague-Grundy theorem, that
+`G` is equivalent to `nim (grundy_value G)`.
```
Comment (semorrison) [inline Mathlib/SetTheory/Game/Nim.lean:21]:
```suggestion
`G` is equivalent to `nim (grundyValue G)`.
```
Please check the other comments.

File: Mathlib/SetTheory/Game/Nim.lean [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+@[simp]
+theorem grundyValue_neg (G : PGame) [G.Impartial] : grundyValue (-G) = grundyValue G := by
+  rw [grundyValue_eq_iff_equiv_nim, neg_equiv_iff, neg_nim, ← grundyValue_eq_iff_equiv_nim]
+#align pgame.grundy_value_neg PGame.grundyValue_neg
+
+theorem grundyValue_eq_mex_right :
+    ∀ (G : PGame) [G.Impartial],
+      grundyValue G = Ordinal.mex.{u, u} fun i => grundyValue (G.moveRight i) := by
+  -- Porting note: was
+  -- | ⟨l, r, L, R⟩ => by
```
Comment (semorrison) [inline Mathlib/SetTheory/Game/Nim.lean:349]:
```suggestion
  | ⟨l, r, L, R⟩, _ => by
```
You can just use this.

File: Mathlib/SetTheory/Game/Nim.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/SetTheory/Game/Nim.lean]: check and didn't find more problems.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**